### DISCUSSION TITLE
Nulleinspeisung NUR bei Batterie?

### DISCUSSION THREAD

**Discussion by 3joern** [hoylabs/OpenDTU-OnBattery#853] [Q&A]
Ahoi Zusammen,
auf die Gefahr hin, das dies schon angesprochen wurde, ich aber nichts konkretes dazu in der Suche gefunden habe, hier nun mein Anliegen.
Inzwischen habe ich ein optimales Setting für den Fertigspeicher von Hornbach (Bluepalm 2500 V1) und meinem 600W BKW mit Südausrichtung. Das eine Modul geht direkt in den HM600 und somit ins Hausnetz, das andere Modul läd den Speicher. So hab ich keine komischen Abschaltungen oder MPPT Überbrückungen, die 50% ins Nirvana schießen. 
Nun möchte ich aber, dass Tagsüber das Modul 1 ohne Power Limiter aka Begrenzung direkt alles was geht ins Hausnetz abgibt. Erst sobald die Sonne weg ist und die Batterie ins Spiel kommt, sollte der Powerlimiter seinen Dienst tun. Gründe hatte ich schon viele (auch hab ich nicht das Gefühl was zu verschenken, möchte nur "optimal" nutzen, was ich aber bei meinem aktuellen Setting darüber hinaus eh nicht kann), aber nachdem ich vorher dieses Video gesehen habe, fand ich es noch schlüssiger: https://www.youtube.com/watch?v=wxP5uMdtMmk

Nun ich kann mir gut vorstellen, dass der Wechselrichter das selbstständig nicht unterscheiden könnte und eben OpenDTUoB auch keine Gedanken lesen kann, daher wäre mein Vorschlag beim Powerlimiter sowas zu ergänzen wie: 
Startzeit Powerlimiter:
Stopzeit Powerlimiter:      
und nur in dieser Zeit wird die Nulleinspeisung realisiert. Das alle 2 Monate anzupassen (wegen Sonnenstand) wäre ja vertretbar. 
Oder gibt es eine andere Möglichkeit, die ich gar nicht sehe?
Habe eh das Phänomen, dass beim eingestellten Limit von 300W der WR durch das Modul alles richtig abgibt, die Batterie Abends aber bei 150W Schluss macht- was aber bei meinem Grundverbrauch nachts optimal ist.

**Comment by spcqike** [comment]
hast du es denn einfach mal mit DPL laufen lassen? Solarpaneel an Eingang1, Batterie an Eingang2. Die Batterie ist ja per Software aus und gibt den Eingang erst zu später Stunde frei.

Am Wechselrichter sollte also tagsüber Leistung an Eingang1 anliegen, an Eingang2 jedoch nicht. der DPL sollte eim HM-600 dynamisch skalieren (da der HM-600 ja einen Eingang pro MPPT hat) und das übertragene Limit damit verdoppeln.
Brauchst du 100W zu Hause, würde er entsprechend 200W vorgeben. sodass Eingang1 alleine die benötigten 100W erzeugen kann.
gleiches Nachts. 150W für das Haus? Eingang1 liefert 0 (weil nacht) -> das Limit sollte 300W betragen, damit Eingang2 die 150W liefern kann.

Hast du das einfach mal probiert?

> Das alle 2 Monate anzupassen (wegen Sonnenstand) wäre ja vertretbar

Wäre das überhaupt notwendig? die OpenDTU kennt den Sonnenstand, berechnet anhand der Position (Breitengrad/Längengrad). Du kannst Sonnenauf- und -untergang bereits auswerten.

Dennoch halte ich dies vor nicht sinnvoll. Der Mischbetrieb aus PV und Batterie ist .... schwierig. 

Was ich begrüßen würde wäre eine andere Art der Skalierung. Eine, die zusätzlich "überskaliert", wenn das gewünschte AC Limit nicht erreicht wird aber mindestens ein Eingang durch das gesetzte Limit gedrosselt wird (potentiell also mehr liefern könnte, als er grade tut)
https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/819

Das würde wahrscheinlich auch bei deinem Aufbau helfen und dazu führen, dass du immer beide Eingänge nutzen kannst.

Wenn du deine Batterie bspw. 18 Uhr zuschalten willst (was ja von der Batterie/App her kommen muss), tust du es einfach. Das PV Paneel könnte noch 50W liefern, du willst aber 130W erzeugen. 

der aktuelle "Skalierer" arbeitet ja wie folgt:
Szenario 1. nur PV, Hausbedarf 130W
PV 50W, batterie aus -> 260W Limit --> 130W pro EIngang -> AC Output 50W
Szenario 2, PV und Batterie
PV 50W, Batterie "unendlich" -> Limit 130W --> 65W pro Eingang -> PV 50W, Batterie 65W -> 115W gesamt output

ein "überskalierender" DPL würde wie folgt regeln können
PV 50, Batterie "unendlich" -> 160W Limit zum Wechselrichter -> 80W pro Eingang. 50W aus der PV (Sonnenbegrenzt) und 80W aus der Batterie (Limiterbegrenzt) -> 130W AC Output

ich sehe diesen Limiter aktuell eher als Feature und nice-2-have für Solar-only Systeme, da er auch einige Nachteile mit sich bringt. Die Nachteile sind im reinen Solarbetrieb jedoch eher nebensächlich, dafür würde man bei unterschiedliche Verschattung der Module halt am Ende mehr kWh herausbekommen.
 
Deine Batterie ist ja, irgendwie, "extern" und vom Hoymiles eigentlich wie ein Solarmodul zu sehen, oder? die tatsächliche Batteriespannung und damit der Schutz vor Tiefenentladung kann ja von der openDTU in deinem Fall gar nicht gewährleistet werden.

**Reply by 3joern** [reply]
Huiii, hab vielen Dank für Dein ausführliches Feedback. Ich glaube aber das Problem was ich habe, konnte ich nicht so deutlich rüber bringen. 

> hast du es denn einfach mal mit DPL laufen lassen? Solarpaneel an Eingang1, Batterie an Eingang2. Die Batterie ist ja per Software aus und gibt den Eingang erst zu später Stunde frei.

Das ist die aktuelle Einstellung, die genau das macht, was sie soll. Tagsüber Solar ins Haus, sobald die Sonne nur noch  50 W bringt, schaltet die Batterie DAZU und übernimmt dann später den Rest begrenz bis max 150W aber abhängig vom Hausverbrauch. Also in der Nacht teilweise nur zwischen 80W und 110W.
Ich möchte aber, das Tagsüber die Sonne nicht Limitiert wird, sondern alles einspeißt, was es gibt - eben auch verschenkt.

Zur Veranschaulichung:
![Bildschirmfoto 2024-04-08 um 17 59 44](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/45c79806-84ff-432f-9f3b-50b5c17809a0)
Früher hatte ich das so, da hat der Speicher aber spezielle Macken (nur die halbe Leistung kam am WR an, Erklärung von jemand mit dem gleichen Speicher: 

> Der Bezugspunkt des WRs liegt auf Plus, entkoppelt durch Dioden. Damit fliessen 1 A vom Plus des 1. Eingang quasi hintenrum in den 2. Eingang. Weil jetzt die beiden Minus- Ausgänge durch Speicher gebrückt sind, arbeitet jetzt auch die 2. Wechselrichterhälfte und speisst 50 Watt aus. Das Ausschalten des 2 Ausgangs wird quasi Ignoriert, Da aber am 2 Eingang keine Spannung vom Speicher kommt, misst der WR am 2 Eingang null Volt und unterschlägt die erzeugt Leistung.
_Und kommt in meinem Fall nirgends an._

Lösung:

```
1 Bluepalm V1 (oder vergleichbar), 2 Panel, HM600 und OpenDTU on Batterie.

1 Panel an PV2, 1 Panel direkt an den WR. Out2 an den WR.
Das Panel am WR deckt die Grundlast ab. Manchmal auch etwas mehr, das verliert man halt, ist in der Praxis fast nicht.

Mittlere Tag:
Auf "Laden vor den Entladen" setzen. Entladestellung auf 5 Watt
Ein Panel macht Grundlast, ein Panel lädt bis die Sonne ganz weg ist, dann wird in die Nacht rein entladen.
Wird dabei der Speicher zu früh voll, dann: Szenario 2

Sehr guter Tag:
Auf "Laden vor den Entladen" setzen. Entladeschwelle 500 Watt
Jetzt gibt der Speicher soviel ab, wie der WR verlangt, der Rest geht in den Speicher. Bei Abschattungen etc. wird aus dem Speicher überbrückt. Auf diese Weise lädt sich der Speicher langsamer, da er schon tagsüber probiert, den Bezug zu minimieren.

Sehr schlechter Tag:
Passtrough aktivieren, dann gehen beide Panel direkt. Die Minileistung würde sonst im Speicher verpuffen.
```
![Bildschirmfoto 2024-04-08 um 17 59 58](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/054f9a9d-73e7-4b94-8b2c-087760995069)

![Bildschirmfoto 2024-04-08 um 18 05 32](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/18442198/b3f6aa3e-5512-41ff-9c54-29dfb1d5e21a)

Das habe ich nun so umgesetzt und es gibt keine Fehler mehr. Nun wird aber Tagsüber kein Überschuss mehr ins allgemeine Stromnetz eingespeist, da der DPL ja den WR begrenzt. Möchte aber das es bei Sonne keine Begrenzung gibt.

**Comment by greymda** [comment]
if i understood correctly the ask, i would disable all the limiters in opendtu-onbattery and use this script created by @reserve85 https://github.com/reserve85/HoymilesZeroExport that runs by cron only during your preferred time (Startzeit Powerlimiter and Stopzeit Powerlimiter).
hope this makes sense.
